pkg/istioversions/versions.go

```diff
@@ -0,0 +1,142 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioversions
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v3"
+
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	//go:embed *.yaml
+	versionsFiles embed.FS
+
+	versionsFilename = "versions.yaml"
+)
+
+// Versions represents the top-level structure of versions.yaml
+type Versions struct {
+	Versions []VersionInfo `json:"versions"`
+}
+
+// AliasInfo contains information about version aliases
+type AliasInfo struct {
+	Name string `json:"name"`
+	Ref  string `json:"ref"`
+}
+
+// VersionInfo contains information about a specific Istio version
+type VersionInfo struct {
+	Name    string          `json:"name"`
+	Ref     *semver.Version `json:"ref"`
+	Version *semver.Version `json:"version"`
+	Repo    string          `json:"repo"`
+	Branch  string          `json:"branch,omitempty"`
+	Commit  string          `json:"commit"`
+	Charts  []string        `json:"charts,omitempty"`
+}
+
+var (
+	// List contains all supported versions. Does not include aliases
+	List []VersionInfo
+	// Map contains version info mapped by version name. Includes mappings for aliases
+	Map map[string]VersionInfo
+	// Default is the default version
+	Default string
+	// Old is the previous supported version
+	Old string
+	// New is the latest supported version
+	New string
+	// AliasList is the alias for the version
+	AliasList []AliasInfo
```

Comment:
I don't think we need this anymore.

Reply:
But why have code just for test purposes if the test can easily find the aliases by itself?

Reply:
I agree it shouldn't be there, but let's fix it in a follow-up, I'd like to produce some RC builds